core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts

```diff
@@ -951,6 +1007,198 @@ export class LoggedOutUser extends BaseUser {
+   * and check if it opens the Math Classroom page.
+   */
+  async clickBrowseOurLessonsButtonInTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      browseOurLessonsButtonInTeachPage,
+      'Browse Our Lessons button',
+      mathClassroomUrl,
+      'Math Classroom'
+    );
+  }
+
+  /**
+   * Function to click the Access Android App button in the Teach page
+   * and check if it opens the Android page.
+   */
+  async clickAccessAndroidAppButtonInTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      accessAndroidAppButtonInTeachPage,
+      'Access the Android App button',
+      androidUrl,
+      'Android'
+    );
+  }
+
+  /**
+   * Function to click the Visit Classroom button in the Teach page
+   * and check if it opens the Math Classroom page.
+   */
+  async clickVisitClassroomButtonInTeachPage(): Promise<void> {
+    await this.page.waitForSelector(visitClassroomButtonInTeachPage);
+    await this.clickButtonToNavigateToNewPage(
+      visitClassroomButtonInTeachPage,
+      'Visit Classroom button',
+      mathClassroomUrl,
+      'Math Classroom'
+    );
+  }
+
+  /**
+   * Function to click the Browse Library button in the Teach page
+   * and check if it opens the Community Library page.
+   */
+  async clickBrowseLibraryButtonInTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      browseLibraryButtonInTeachPage,
+      'Browse Library button',
+      communityLibraryUrl,
+      'Community Library'
+    );
+  }
+
+  /**
+   * Function to click the Browse Our Lessons button in the Teach page
+   * and check if it opens the Math Classroom page.
+   */
+  async clickExploreLessonsButtonInTeachPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      exploreLessonsButtonInTeachPage,
+      'Explore Lessons button',
+      mathClassroomUrl,
+      'Math Classroom'
+    );
+  }
+
+  /**
+   * Function to click the Partner With Us button in the Partnerships page
+   * and check if it opens the Partnerships Google form.
+   * The button is in the first section of the page.
+   */
+  async clickPartnerWithUsButtonInPartnershipsPage(): Promise<void> {
+    await this.clickButtonToNavigateToNewPage(
+      partnerWithUsButtonAtTheTopOfPartnershipsPage,
+      'Partner With Us button at the top of the Partnerships page',
+      partnershipsFormUrl,
+      'Partnerships Google Form'
+    );
+  }
+
+  /**
+   * Function to change the site language to the given language code.
+   * @param langCode - The language code to change the site language to. Example: 'pt-br', 'en'
+   */
+  async _changeSiteLanguage(langCode: string): Promise<void> {
+    const languageDropdown = '.e2e-test-language-dropdown';
+    const languageOption = `.e2e-test-i18n-language-${langCode} a`;
+    await this.clickOn(languageDropdown);
+    await this.clickOn(languageOption);
+    await this.page.waitForTimeout(2000);
```

Comment:
Do not use waitForTimeout here and elsewhere. Wait for a particular thing to happen instead. Otherwise we just end up with flaky tests.